### PR TITLE
Fix BACKUP TO S3 for Google Cloud Storage

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `BACKUP TO S3` for Google Cloud Storage

GCS does not support batch delete, regular S3 disk has a check for batch delete and a fallback to regular delete if it not supported, but for BACKUPs I think that it is OK to use non-batch delete always.

Follow-up for: #42333 (cc @vitlibar )

### CODE REVIEW

**Review by alesapin — COMMENTED** [review]
Why we cannot implement the same check that we have in DiskS3?

**Comment by azat** [comment]
>Why we cannot implement the same check that we have in DiskS3?

It does not looks like it worth to add the check here, since the only places where there is file deletion during BACKUP is:
- rollback in case of failure
- and removing the lock file in regular path

Objections?

**Comment by azat** [comment]
Plus, AFAICS delete requests is free of charge in [GCS](https://cloud.google.com/storage/pricing) and [S3](https://aws.amazon.com/s3/pricing/)

**Comment by azat** [comment]
>To avoid thousands of synchronous S3 requests in case of backup failure?

Ok. Fixed.

File: src/Backups/BackupIO_S3.cpp [hunk 1]
```diff
@@ -385,6 +404,58 @@ void BackupWriterS3::removeFiles(const Strings & file_names)
...
+        LOG_TRACE(log, "Batch delete: supported");
+        supports_batch_delete = true;
+    }
+    catch (const Exception &)
+    {
+        removeFilesSingle({path});
+
+        LOG_TRACE(log, "Batch delete: unsupported");
+        supports_batch_delete = false;
+    }
```
Comment (vitlibar) [inline src/Backups/BackupIO_S3.cpp:456]: This is quite a difficult way to check if batch delete is supported. Maybe for `s3` disks it's ok, however it doesn't look ok here. First, if the backup was successful you have to remove only one file - and if there is one file you have no reason to use batch delete anyway (and no need to check anything too). Second, if it's necessary to remove more than one file it's still no reason to check anything in advance - just try to call removeFilesBatch() and only if it's failed - check the error code and then set `supports_batch_delete` to 0 and call `removeFilesSingle()`.

**Comment by azat** [comment]
>230 successful and 1 skipped checks

Haven't seen this for awhile...

File: src/Backups/BackupIO_S3.cpp (hunk 1 above)
Comment (azat) [reply src/Backups/BackupIO_S3.cpp]: Ok. Rebased.

**Review by vitlibar — APPROVED** [review]

**Pull request merged**